packages/reakit/src/Menu/Menu.tsx

```diff
@@ -0,0 +1,40 @@
+import React from "react";
+import { ifNotProp, theme } from "styled-tools";
+import styled from "../styled";
+import BaseList from "../List";
+import Navigation from "../Navigation";
+import withPropAt from "../_utils/withPropAt";
+
+export interface MenuProps {
+  horizontal?: boolean;
+  horizontalAt?: number | string;
+}
+
+const horizontalAt = withPropAt("horizontalAt");
+const HorizontalMenuContext = React.createContext(false);
+
+const List = styled(BaseList)`
+  display: flex;
+  flex-wrap: wrap;
+  flex-direction: ${ifNotProp("horizontal", "column", "row")};
+  ${horizontalAt("flex-direction: row")};
+
+  ${theme("Menu")};
+`;
+
+export { HorizontalMenuContext };
+
+export default function Menu(props: MenuProps) {
+  const layout = {
+    horizontal: Boolean(props.horizontal),
+    horizontalAt: props.horizontalAt
+  };
+
+  return (
+    <HorizontalMenuContext.Provider value={layout}>
```

Comment:
We're already spreading the props to `<List />` which is the lowest level child `<Menu />` knows about:
```jsx
function Menu(props: MenuProps) {
   // ...
   <List {...props} />
   // ...
}
```
To make it work the way you propose we would need to modify `<List />` so that it also spreads its props down to its `children`.

However that would imply tying the behaviour of one component to the other and end with tight coupling between the two 🧐 👎🏻

Reply:
Yeah yeah... big nono 💥

Reply:
Let's try to render only one component. Context isn't needed. Just add [aria-orientation](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-orientation_attribute) and style its children based on it. For example:
```
const MenuDivider = styled(Box)`
  [aria-orientation="horizontal"] > & {
    // styles for vertical
  }
`;
```